### PR TITLE
ceph-volume: lvm: get_device_vgs() filter by provided prefix

### PR DESCRIPTION
**Pull request by yehudasa**
Fixes: https://tracker.ceph.com/issues/44246

Signed-off-by: Yehuda Sadeh <yehuda@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by liewegas** [comment]
retest this please

**Comment by jan--f** [comment]
jenkins test ceph-volume tox

**Comment by jan--f** [comment]
I think this looks fine.
However I'd like to reproduce this, as I suspect this only covers up another issue. As I mentioned on the tracker ticket, I can't reproduce this by calling `prepare --bluestore --no-systemd` on a blank drive or a drive that has a vg on it. Is there any specific environment I need to setup?

**Comment by liewegas** [comment]
passes cephadm tests, fwiw http://pulpito.ceph.com/sage-2020-02-25_12:35:01-rados-wip-sage-testing-2020-02-24-2147-distro-basic-smithi/

**Comment by jan--f** [comment]
The change here is fine but I doubt it fixes the reported bug. With this change `get_device_vgs` returns less then before. In the tracker ticket `lvcreate` fails cause it is called without a vg name. I don't see how this change here should fix that.

File: src/ceph-volume/ceph_volume/api/lvm.py [hunk 1]
```diff
@@ -866,7 +866,7 @@ def get_device_vgs(device, name_prefix=''):
         verbose_on_failure=False
     )
     vgs = _output_parser(stdout, VG_FIELDS)
-    return [VolumeGroup(**vg) for vg in vgs]
+    return [VolumeGroup(**vg) for vg in vgs if vg['vg_name'].startswith(name_prefix)]
```
Comment (jan--f) [inline src/ceph-volume/ceph_volume/api/lvm.py]:
```suggestion
    return [VolumeGroup(**vg) for vg in vgs if vg['vg_name'] and vg['vg_name'].startswith(name_prefix)]
```

Comment (jan--f) [reply src/ceph-volume/ceph_volume/api/lvm.py]: We also need to avoid creating a VG with an empty name.

**Comment by jan--f** [comment]
jenkins test ceph-volume tox

**Comment by liewegas** [comment]
retest this please

**Pull request merged**